.github/workflows/cmake-ctest.yml

```diff
@@ -7,41 +7,51 @@ on:
     branches: [ "master" ]
 
 env:
-  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
   BUILD_TYPE: Release
-  WITH_DLT_UNIT_TESTS: ON
+  WITH_DLT_COVERAGE: ON
   BUILD_GMOCK: OFF
 
 jobs:
   build:
-    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
-    # You can convert this to a matrix build if you need cross-platform coverage.
-    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v3
 
     - name: Prepare submodule
       run: |
-          git submodule init 
+          git submodule init
           git submodule update
 
     - name: Configure CMake
-      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a
-      # single-configuration generator such as make.
-      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
       run: |
           cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} \
-                                               -DWITH_DLT_UNIT_TESTS=${{env.WITH_DLT_UNIT_TESTS}} \
+                                               -DWITH_DLT_COVERAGE=${{env.WITH_DLT_COVERAGE}} \
                                                -DBUILD_GMOCK=${{env.BUILD_GMOCK}}
 
     - name: Build
-      # Build your program with the given configuration
       run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
 
     - name: Test
       working-directory: ${{github.workspace}}/build
-      # Execute tests defined by the CMake configuration.
-      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
       run: ctest -C ${{env.BUILD_TYPE}}
+
+    - name: Install dependencies
+      run: sudo apt-get install lcov
+
+    - name: Generate lcov report
+      working-directory: ${{github.workspace}}
+      run: bash util/dlt_coverage_report/lcov_report_generator.sh build -xe
+
+    - name: Upload lcov report
+      if: github.ref == 'refs/heads/master'
```

Comment:
Please install dependencies and generate lcov report when it is also uploaded. Otherwise, checks for PRs will take unnecessarily long.

Reply:
Please check if the results can be stored as an build artifact instead. This would avoid the constant uploads and artifiacts will be deleted after 90 days. No reason to keep this historic data longer.

Reply:
Hello @michael-methner ,
Artifact is available now.
Whenever there is a PR, an Artifact for lcov report is created.

Whenever there is a Merge from PR to master branch, Artifact available and result updated on branch dlt_coverage_report.

The branch is necessary for constructing an online interactive webpage without merging any new lcov dir onto master.

Please kindly review!